e2e-tests/tfm/cypress/e2e/journeys/deals/deals-pagination.spec.js

```diff
@@ -0,0 +1,138 @@
+import relative from '../../relativeURL';
+import pages from '../../pages';
+import { T1_USER_1 } from '../../../../../e2e-fixtures';
+
+context('User can navigate through a paginated table of deals using the pagination navigation', () => {
+  const numberOfDeals = 62;
+
+  before(() => {
+    cy.deleteAllTfmDealsFromDb();
+    cy.deleteAllTfmDealsFromDb();
+    cy.insertManyTfmDealsIntoDb(numberOfDeals);
+  });
+
+  beforeEach(() => {
+    cy.login(T1_USER_1);
+
+    cy.url().should('eq', relative('/deals/0'));
+    cy.checkDealIdCell({ rowNumber: 0, expectedDealId: '10000001' });
+    cy.checkDealIdCell({ rowNumber: 19, expectedDealId: '10000020' });
+    cy.checkDealsTableRowsTotal(20);
+  });
+
+  after(() => {
+    cy.deleteAllTfmDealsFromDb();
+  });
+
+  it('should allow the user to navigate to the next page of the deals table', () => {
+    pages.dealsPage.pagination.next().click();
+
+    cy.url().should('eq', relative('/deals/1'));
+    cy.checkDealIdCell({ rowNumber: 0, expectedDealId: '10000021' });
+    cy.checkDealIdCell({ rowNumber: 19, expectedDealId: '10000040' });
+    cy.checkDealsTableRowsTotal(20);
+  });
+
+  it('should allow the user to navigate to the last page of the deals table', () => {
+    pages.dealsPage.pagination.last().click();
+
+    cy.url().should('eq', relative('/deals/3'));
+    cy.checkDealIdCell({ rowNumber: 0, expectedDealId: '10000061' });
+    cy.checkDealIdCell({ rowNumber: 1, expectedDealId: '10000062' });
+    cy.checkDealsTableRowsTotal(2);
+  });
+
+  it('should allow the user to navigate to the previous page of the deals table', () => {
+    cy.visit('/deals/3');
+    pages.dealsPage.pagination.previous().click();
+
+    cy.url().should('eq', relative('/deals/2'));
+    cy.checkDealIdCell({ rowNumber: 0, expectedDealId: '10000041' });
+    cy.checkDealIdCell({ rowNumber: 19, expectedDealId: '10000060' });
+    cy.checkDealsTableRowsTotal(20);
```

Comment:
I'm wondering if we need to check each row/deal ID rather than 2 of 20, maybe could have something like

```js
cy.checkDealsTableRows({ dealIds, totalRows = 20 });
```

And inside this command can call `checkDealIdCell()` X times and increment from a single `dealId` or passed in deal IDs. What do you think?

Thinking this could help to ensure initial sorting/ordering is working as expected? Not sure if this is handled somewhere else

Reply:
Good shout, I've replaced with a command of this form:
`cy.checkDealIdCells({ firstDealId: '10000001', increment: 1, numberToCheck: 20 });`